File: clients/soa_services/parameter/python/src/crosslab/soa_services/parameter/messages.py
from typing import List, Literal, TypedDict


class ParameterDescription(TypedDict):
    name: str
    unit: str
    minimum: float
    maximum: float


class ParameterConfig(ParameterDescription):
    remoteName: str


class ParameterChangedEvent(TypedDict):
    parameter: str
    value: float


class ParameterListChangedEvent(TypedDict):
    parameters: List[ParameterDescription]


class ParameterServiceConfig(TypedDict):
    serviceType: Literal["http://api.goldi-labs.de/serviceTypes/parameter"]
    parameters: List[ParameterConfig]

File: clients/soa_services/parameter/js/src/messages.ts
export interface ParameterDescription {
  name: string;
  unit: string;
  minimum: number;
  maximum: number;
}

export interface ParameterServiceSetupEvent {
  parameters: ParameterDescription[];
}

export interface ParameterServiceEvent {
  parameter: string;
  value: number;
}
